pkg/signup/signup_service.go

```diff
@@ -240,25 +242,46 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 }
 
 // GetUserSignup is used to return the actual UserSignup resource instance, rather than the Signup DTO
-func (s *ServiceImpl) GetUserSignup(userID string) (*crtapi.UserSignup, error) {
+func (s *ServiceImpl) GetUserSignup(userID string) (*v1alpha1.UserSignup, error) {
 	// Retrieve UserSignup resource from the host cluster
 	userSignup, err := s.UserSignups.Get(userID)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return nil, nil
-		}
 		return nil, err
 	}
 
 	return userSignup, nil
 }
 
 // UpdateUserSignup is used to update the provided UserSignup resource, and returning the updated resource
-func (s *ServiceImpl) UpdateUserSignup(userSignup *crtapi.UserSignup) (*crtapi.UserSignup, error) {
+func (s *ServiceImpl) UpdateUserSignup(userSignup *v1alpha1.UserSignup) (*v1alpha1.UserSignup, error) {
 	userSignup, err := s.UserSignups.Update(userSignup)
 	if err != nil {
 		return nil, err
 	}
 
 	return userSignup, nil
 }
+
+func (s *ServiceImpl) CheckIfUserIsKnown(userID, countryCode, phoneNumber string) error {
+	bannedUserList, err := s.BannedUsers.ListByPhoneNumber(countryCode + phoneNumber)
+	if err != nil {
+		return errors3.NewInternalError(err, "failed listing banned users")
+	}
+	for _, bannedUser := range bannedUserList.Items {
+		if bannedUser.Name == userID {
```

Comment:
If the phone number exists _at all_ in BannedUser then the user should not be able to proceed, so this check seems unnecessary.

Reply:
ohhh okay, so what i had earlier should be fine, i've updated as part of this commit: 4ebb9c3029dabcb8590ae9e0e0984c3b6f348b7d